model/DataObject.php

```diff
@@ -781,7 +781,15 @@ public function plural_name() {
 			return $name;
 		} else {
 			$name = $this->singular_name();
-			if(substr($name,-1) == 'y') $name = substr($name,0,-1) . 'ie';
+			//if it ends in a y, then it's a special case
+			if(substr($name, -1) == 'y') {
+				$vowels = array('a', 'e', 'i', 'o', 'u');
```

Comment:
I'd be surprised if regexing a single char was faster than an `in_array` though I'd rather do the fastest option.

somthing like `[^aeiou]y$` ?